packages/front-end/pages/datasources/[did].tsx

```diff
@@ -59,7 +58,7 @@ const DataSourcePage: FC = () => {
 
   const canEdit =
     (d &&
-      checkDatasourceProjectPermissions(d, permissions, "createDatasources") &&
+      permissions.check("editDatasourceSettings", d.projects) &&
```

Comment:
Is this change intentional?  Changing the permission that's being checked

Reply:
It was - to me this feels like the more appropriate permission to check.

Checking against `createDatasources` mean that only `admins` will be able to edit a data source.

Looking at all of the components of the `[did]` page (things like "ExperimentAssignmentQueries`, `DataSourceInlineEditIdentifierTypes`, etc - those all check for against `editDatasourceSettings`. So having this check against `createDatasources` means we never rended the `Edit` button, but if the user was able to open the components, they would then have the ability to edit.

Reply:
I've dug into this more, and I think this deserves its own PR. We've got two permissions around data sources `createDatasources` and `editDatasourceSettings`. 

If a user is to create a data source or update an existing datasource's connection settings, that should require a `createDatasources` permission. But updating the data source name, projects, and other settings (excluding the connection settings) should only require the `editDatasourceSettings`. 

Currently, the modal to edit a data sources name, projects, and description is in the same form as the connection settings, so even if the `experimenter` user isn't updating the connection settings, they get a permissions error because `params` are included in the body and thus, the permission being checked is `createDatasources`.

It seems like we need to update [did] to specifiy editing connection settings and deleting, vs editing connection info. Likewise, we should give the user the ability to edit projects/name/description outside of the "Edit Connection Info" modal

Reply:
Another update - I updated [did].tsx to differentiate between if the user can edit or if they can create/delete. So now, if a user can edit a datasource's settings, they can do so, but we only show `admins` the "Edit Connection Info" and "Delete" buttons. 

We should probably have a convo about whether a `Experimenter` can edit the name, description, or projects for a data source. Editing the projects could affect other people's permissions, so I could see an argument that only admins can do that.